### PR TITLE
Antagonist Critter Spawn admin customization options

### PR DESCRIPTION
**Pull request by Sovexe**
## About the PR 
The Antagonist Critter Spawn event now allows for customization in the form of critter type and quantity.

demo: https://streamable.com/qi6xy6

## Why's this needed? 
Fun.

### CODE REVIEW

**Comment by Sovexe** [comment]
converting to a draft temporarily as I think I may have found a bug with this

File: code/modules/events/antagonist_critter_ghost_respawn.dm [hunk 1]
```diff
@@ -18,15 +20,34 @@
 		if (..())
 			return
 
-		var/input = input(usr,"Which one? Pick null for random???",src.name) as null|anything in list("fire_elemental","spider","gunbot")
-		if (!input || !istext(input))
+		var/input = input(usr,"Which one? Press cancel for random.",src.name) as null|anything in list("fire_elemental","spider","gunbot","custom critter path")
```
Comment (mybluecorners) [inline code/modules/events/antagonist_critter_ghost_respawn.dm:23]: you can outright remove the options here, just let them choose between custom and random

File: code/modules/events/antagonist_critter_ghost_respawn.dm [hunk 2]
```diff
@@ -55,17 +76,25 @@
...
 			var/atom/pestlandmark = pick(EV)
 
-			var/list/select = pick(pest_invasion_critter_types)
-			if (source)
-				if (source == "fire_elemental")
+			var/list/select = null
+			switch (source)
+				if (null) //random
+					select = pick(src.pest_invasion_critter_types)
+				if ("fire_elemental")
```
Comment (mybluecorners) [inline code/modules/events/antagonist_critter_ghost_respawn.dm:83]: same down here, remove the fire_elemental spider and gunbot checks

Comment (Sovexe) [reply code/modules/events/antagonist_critter_ghost_respawn.dm]: I have made the requested changes

**Pull request merged**